tests/headline_grabber/pipeline_steps/test_scrape_text_wap.py

```diff
@@ -0,0 +1,148 @@
+    print("***CCCC****")
+
+    assert actualResult.description.strip() == "While critics may worry that he is cocooned from the worst of the news, living in a bit of a bubble, President Biden has chosen to focus on the positives."
+    assert actualResult.title.strip() == "Biden is seeing a different world than other Democrats"
+    assert actualResult.link.strip() == 'https://www.washingtonpost.com/politics/2024/07/08/biden-different-world-bubble/'
+
+
+def test_filter_results_False():
+    scrape_text = ScrapeText()
+    link = "https://www.washingtonpost.com/politics/2024/07/08/biden-different-world-bubble/"
+    description = "hello"
+    title = "hello"
+    headline = Headline(link, title, description)
+    assert scrape_text._filter_results(headline) is False
+    print("***EEEE****")
+
+    empty_link = ""
+    headline = Headline(empty_link, title, description)
+    
+    assert scrape_text._filter_results(headline) is False
+    print("***FFFF****")
+
+    long_title = "Biden is seeing a different world than other Democrats"
+    long_description = "While critics may worry that he is cocooned from the worst of the news, living in a bit of a bubble, President Biden has chosen to focus on the positives."
+   
+    headline = Headline(empty_link, long_title, long_description)
+    
+    assert scrape_text._filter_results(headline) is False
+    print("***GGGG****")
+
+
+def test_filter_results_True():
+    print("***HHHH****")
+
+    scrape_text = ScrapeText()
+    link = "https://www.washingtonpost.com/politics/2024/07/08/biden-different-world-bubble/"
+    description = "While critics may worry that he is cocooned from the worst of the news, living in a bit of a bubble, President Biden has chosen to focus on the positives."
+    title = "Biden is seeing a different world than other Democrats"
+    headline = Headline(link, title, description)
+    
+    print("***II****")
+
+    assert scrape_text._filter_results(headline) is True
+    print("***JJ****")
+
+
+def test_run_Exception():
+    scrape_text = ScrapeText()
+    config = ScrapeTextWap.ERROR_CONFIG
+    pipeline_context = PipelineContext(
+        site_configs=[config],
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={},
+        user_input=UserPreferences(
+            include=['bgb'],
+            exclude=None,
+        ),
+    )
+    with pytest.raises(ScrapeTextException) as exceptionInfo:
+        scrape_text.run(pipeline_context)
+    assert "No headlines found." in str(exceptionInfo)
+
+def test_run_Success():
+    scrape_text = ScrapeText()
+    config = ScrapeTextWap.WAP_CONFIG
+    pipeline_context = PipelineContext(
+        site_configs=[config],
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={},
+        user_input=UserPreferences(
+            include=['wap'],
+            exclude=None,
+        ),
+    )
+    actualResult = scrape_text.run(pipeline_context)
```

Comment:
you shouldn't be running the entire step unless you plan to mock the html requests, which you're not. Either mock the html requests, or pass in a stub html string to the get headlines beautifulsoup method